### PR TITLE
support delta eds, currently only support serviceentry

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

**Comment by ramaraochavali** [comment]
Can you please add a test case for it?

**Comment by hzxuzhonghu** [comment]
I thought it depends on adsc https://github.com/istio/istio/issues/36297, after a  look at delta_test.go, it does not use that package.

**Comment by hzxuzhonghu** [comment]
Will add some cases

**Review by howardjohn — COMMENTED** [review]
A bit confused what the benefit is. Don't we already basically do this with EDS with `full=false`, even in SotW?

Is this to make full push more efficient?

**Comment by hzxuzhonghu** [comment]
> Is this to make full push more efficient?

Yes

**Comment by howardjohn** [comment]
Thanks, makes sense now. Will take a look soon after I am back from holidays

**Comment by hzxuzhonghu** [comment]
pig @howardjohn

File: pilot/pkg/xds/eds.go [hunk 1]
```diff
@@ -484,3 +441,138 @@ func buildEmptyClusterLoadAssignment(clusterName string) *endpoint.ClusterLoadAs
...
+			cached++
+		} else {
+			// if a service is not found, it means the cluster is removed
+			if builder.service == nil {
+				removed = append(removed, clusterName)
+				continue
+			}
+			l := eds.Server.generateEndpoints(builder)
+			if l == nil {
+				continue
```
Comment (howardjohn) [inline pilot/pkg/xds/eds.go:560]: does this mean removed as well?

**Review by howardjohn — APPROVED** [review]
LGTM, one comment but can get this in

File: pilot/pkg/xds/eds.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/xds/eds.go:560]: yes, i made this change also to suite stow.

**Pull request merged**